### PR TITLE
Storybook

### PR DESCRIPTION
**Pull request by xasopheno**
Adds Storybook with support for CSS modules. Definition following in my editor does not go to the css definition, but instead the module declarations. Maybe this is just a kink in my setup.

### CODE REVIEW

File: template/packages/frontend/package.json [hunk 1]
```diff
@@ -23,12 +24,19 @@
...
+    "@storybook/react": "^6.0.28",
     "@types/node": "^14.14.6",
     "@types/react": "^16.9.55",
     "@types/react-dom": "^16.9.9",
+    "@types/storybook__react": "^5.2.1",
+    "babel-loader": "^8.2.0",
+    "babel-preset-react-app": "^10.0.0",
     "concurrently": "^5.3.0",
     "eslint-plugin-graphql": "^4.0.0",
+    "storybook-css-modules-preset": "^1.0.2",
```
Comment (xasopheno) [inline template/packages/frontend/package.json:39]: This should be dropped if we opt for another css framework.

File: template/packages/frontend/.storybook/main.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    checkOptions: {},
+    reactDocgen: "react-docgen-typescript",
+    reactDocgenTypescriptOptions: {
+      shouldExtractLiteralValuesFromEnum: true,
+      propFilter: prop =>
+        prop.parent ? !/node_modules/.test(prop.parent.fileName) : true,
+    },
+  },
+  addons: ["storybook-css-modules-preset"],
+  stories: ["../stories/*.stories.@(ts|tsx)"],
```
Comment (hallettj) [inline template/packages/frontend/.storybook/main.js:13]: If you use the recursive glob (`**`) stories don't have to be limited to the stories directory (but that can still be the convention), and we can put stories in subdirectories.

```suggestion
  stories: ["../**/*.stories.@(ts|tsx)"],
```

File: template/packages/frontend/components/Title.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  text: string
+}
+
+const Title = ({ text }: Props) => {
+  return (
+    <div>
+      <h2 className={styles.title}>{text}</h2>
+    </div>
+  )
+}
```
Comment (hallettj) [inline template/packages/frontend/components/Title.tsx:14]: If you use the special `children` prop then the component can be called like, `<Title>My Title</Title>`. The `React.ReactNode` type allows a string, a React element, an array, or a few other types that are allowed to be interpolated in JSX.

```suggestion
type Props = {
  children: React.ReactNode
}

const Title = ({ children }: Props) => {
  return (
    <div>
      <h2 className={styles.title}>{children}</h2>
    </div>
  )
}
```

File: template/packages/frontend/package.json [hunk 4]
```diff
@@ -23,12 +24,20 @@
...
+    "@babel/core": "^7.12.3",
     "@hallettj/eslint-plugin-ts-graphql": "^0.1.1",
+    "@storybook/addon-docs": "^6.0.28",
+    "@storybook/react": "^6.0.28",
     "@types/node": "^14.14.6",
     "@types/react": "^16.9.55",
     "@types/react-dom": "^16.9.9",
+    "@types/storybook__react": "^5.2.1",
+    "babel-loader": "^8.2.0",
+    "babel-preset-react-app": "^10.0.0",
```
Comment (hallettj) [inline template/packages/frontend/package.json:36]: I'm a bit bothered by adding these babel dependencies if we don't directly reference them. Are they necessary? This applies to `@babel/core` too.

**Review by hallettj — APPROVED** [review]
If we go this route I'd like to see a working example with a Sass module.

File: template/packages/frontend/package.json (hunk 4 above)
Comment (xasopheno) [reply template/packages/frontend/package.json:36]: Seems that babel is required for storybook. 
https://github.com/storybookjs/storybook/issues/7156

**Pull request merged**